### DISCUSSION TITLE
alert-buster and disabled console

### DISCUSSION THREAD

**Discussion by MasterKia** [uBlockOrigin/uBlock-issues#2259] [Ideas]
Can sites circumvent `##+js(alert-buster)` by disabling the console log?
> The Web Console logging API (console.log, console.info, console.warn, console.error) has been disabled by a script on this page. 

```js
/// alert-buster.js
(function() {
    window.alert = function(a) {
        console.info(a);
    };
})();
```

**Comment by ghost** [comment]
Alert will be still redirected, but will not be print into console?

**Comment by krystian3w** [comment]
Soft98 detected usage alert-buster?

**Reply by MasterKia** [reply]
I've seen example(s) of `alert-buster` not working before, can't find them now.

**Reply by krystian3w** [reply]
Maybe too high of html document and these can bye named as race condition.

**Reply by MasterKia** [reply]
I wonder why not just set window.alert as noopFunc instead of redirecting to console?

**Reply by krystian3w** [reply]
Also similar looks `confirm()` and `prompt()`

https://www.w3schools.com/jsref/met_win_confirm.asp
https://www.w3schools.com/jsref/met_win_prompt.asp
